superset/utils/dashboard_import_export.py

```diff
@@ -17,33 +17,172 @@
+            dashboard, import_time=import_time)
+
+    # Import any files in this exported Dashboard
+    if 'files' in data:
+        if len(data['files']) > 0:
+            examples_engine = get_or_create_main_db()
+            for table in data['files']:
+                logging.info(f'Import data from file {table["file_name"]} into table ' +
+                             f'{table["table_name"]}')
+                df = pd.read_csv(table['file_name'], parse_dates=True,
+                                 infer_datetime_format=True, compression='infer')
+                df.to_sql(
+                    table['table_name'],
+                    examples_engine.get_sqla_engine(),
+                    if_exists='replace',
+                    chunksize=500,
+                    index=False)
+
+    session.commit()
+
+
+def get_db_name(uri):
+    """Get the DB name from the URI string"""
+    db_name = make_url(uri).database
+    if uri.startswith('sqlite'):
+        db_name = re.match('(?s:.*)/(.+?).db$', db_name).group(1)
+    return db_name
+
+
+def get_default_example_db():
+    """Get the optional substitute database for example import"""
+    uri = app.config.get('SQLALCHEMY_EXAMPLES_URI')
+    db_name = get_db_name(uri)
+
+    return db.session.query(Database).filter_by(
+        database_name=db_name).one()
+
+
+def import_example_dashboard(session, import_example_json, data_blob_urls,
+                             database_uri, import_time=None):
+    """Imports dashboards from a JSON string and data files to databases"""
+    data = json.loads(import_example_json, object_hook=decode_dashboards)
+
+    substitute_db_name = get_db_name(database_uri) or \
+        get_default_example_db().database_name
+
+    for table in data['datasources']:
+        logging.debug(
+            f'Importing table: {table} in substitute_db_name: {substitute_db_name}')
+        type(table).import_obj(table, import_time=import_time,
+                               substitute_db_name=substitute_db_name)
+
     session.commit()
     for dashboard in data['dashboards']:
         Dashboard.import_obj(
             dashboard, import_time=import_time)
+
+    if len(data['files']) > 0:
+        examples_engine = get_or_create_example_db(database_uri)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            for file_info in data['files']:
+                # Get the github info for the file
+                blob_file_path = f'{tmpdir}{os.path.sep}{file_info["file_name"]}'
+                blob_url = data_blob_urls[file_info['file_name']]
+
+                response = requests.get(blob_url, stream=True)
+                with open(blob_file_path, 'wb') as out_file:
+                    shutil.copyfileobj(response.raw, out_file)
+                del response
+
+                df = pd.read_csv(blob_file_path, parse_dates=True,
+                                 infer_datetime_format=True, compression='infer')
+                df.to_sql(
```

Comment:
This section scream deja vue with the section above, now I'm not sure why we need both `import_dashboard` and `import_example_dashboard`, but if we do we shouldn't repeat the logic

Reply:
Fair point, I will factor out what is duplicated.

Reply:
Factored out.